### PR TITLE
[SPARK-18361] [PySpark] Expose RDD localCheckpoint in PySpark

### PR DESCRIPTION
**Pull request by gabrielhuang**
## What changes were proposed in this pull request?

Expose RDD's localCheckpoint() and associated functions in PySpark.

## How was this patch tested?

I added a UnitTest in python/pyspark/tests.py which passes.

I certify that this is my original work, and I license it to the project under the project's open source license.

Gabriel HUANG
Developer at Cardabel (http://cardabel.com/)

### CODE REVIEW

**Comment by andrewor14** [comment]
ok to test

File: python/pyspark/rdd.py [hunk 1]
```diff
@@ -181,6 +181,7 @@ def __init__(self, jrdd, ctx, jrdd_deserializer=AutoBatchedSerializer(PickleSeri
         self._jrdd = jrdd
         self.is_cached = False
         self.is_checkpointed = False
+        self.is_locally_checkpointed = False
```
Comment (andrewor14) [inline python/pyspark/rdd.py]: is this read anywhere?

**Comment by andrewor14** [comment]
Looks good, just one question.

File: python/pyspark/rdd.py (hunk 1 above)
Comment (gabrielhuang) [inline python/pyspark/rdd.py]: No it's not. I just mirrored the behavior of self.is_checkpointed: the flag self.is_locally_checkpointed is set by the Python wrappers, but isLocallyChecked() wraps around the scala equivalent, which I imagine reads itself a scala flag.

**Comment by gabrielhuang** [comment]
There's a sleep(1.) Instruction that causes the test to time out. Is that a problem?

**Comment by viirya** [comment]
Looks like just an IO problem when archiving logs after successful tests.

**Comment by viirya** [comment]
LGTM

**Comment by viirya** [comment]
retest this please.

File: python/pyspark/rdd.py (hunk 1 above)
Comment (andrewor14) [inline python/pyspark/rdd.py]: so we probably don't need it since we're always just calling the scala version

**Comment by gabrielhuang** [comment]
Ok, I removed the Python flag.

**Comment by gabrielhuang** [comment]
retest this please.

**Comment by andrewor14** [comment]
LGTM merging into master thanks.

**Comment by rxin** [comment]
@andrewor14 did you also merge this in branch-2.1?

**Comment by andrewor14** [comment]
I did

**Pull request closed without merging**